bdk_core/src/descriptor_tracker.rs

```diff
@@ -12,81 +12,81 @@ use miniscript::{
     descriptor::DerivedDescriptorKey, psbt::PsbtInputExt, Descriptor, DescriptorPublicKey,
 };
 
+/// A [DescriptorTracker] is a collection of maps and lists, that contains
+/// every thing that needs to be stored to perform syncing and wallet operations
+/// for the all script pubkeys that can be derived from the descriptor.
 #[derive(Clone, Debug)]
 pub struct DescriptorTracker {
     /// The descriptor we are tracking
     descriptor: Descriptor<DescriptorPublicKey>,
-    /// Which txids are included in which checkpoints
-    checkpointed_txs: BTreeMap<u32, _CheckPointData>,
-    /// Index the outpoints owned by this tracker to the derivation index of script pubkey.
+    /// A [CheckPointData] store indexed blockheight.
+    checkpoints: BTreeMap<u32, CheckPointData>,
+    /// Index the Outpoints owned by this tracker to the derivation index of script pubkey.
     txouts: BTreeMap<OutPoint, u32>,
-    /// Which tx spent each output. This indexes the spends for every transaction regardless if the
-    /// outpoint has one of our scripts pubkeys.
+    /// List of all known spends. Including our's and other's outpoints. Both confirmed and unconfirmed.
+    /// This is useful to track all inputs we might ever care about.
     spends: BTreeMap<OutPoint, (u32, Txid)>,
-    /// The unspent txouts
+    /// Set of our unspent outpoints.
     unspent: HashSet<OutPoint>,
-    /// The ordered script pubkeys that have been derived from the descriptor.
-    scripts: Vec<Script>,
-    /// A reverse lookup from script to derivation index
+    /// Derived script_pubkeys ordered by derivation index.
+    derived_scripts: Vec<Script>,
+    /// A reverse lookup from out script_pubkeys to derivation index
     script_indexes: HashMap<Script, u32>,
-    /// A lookup from script pubkey to outpoint
+    /// A lookup from script pubkey derivation index to related outpoints
     script_txouts: BTreeMap<u32, HashSet<OutPoint>>,
-    /// A set of script derivation indexes that haven't been spent to
+    /// A set of unused derivation indices.
     unused: BTreeSet<u32>,
-    /// Map from txid to metadata
+    /// A transaction store of all potentially interesting transactions. Including ones in mempool.
     txs: HashMap<Txid, AugmentedTx>,
-    /// Index of transactions that are in the mempool
+    /// A list of mempool [Txid]s
     mempool: HashSet<Txid>,
-    // TODO change to blocktime + height
-    // Optionally we need the consensus time i.e. Median time past
-    // https://github.com/bitcoin/bitcoin/blob/a4e066af8573dcefb11dff120e1c09e8cf7f40c2/src/chain.h#L290-L302
-    latest_blockheight: Option<u32>,
+    /// The latest known block heigh + time.
+    latest_blocktime: Option<BlockTime>,
```

Comment:
Hmm actually this should not have the height at all. the `new_tip` is the height...
It should just be called `tip_time` and be a u32.